Sources/Sandbox.Game/Game/Entities/MyEntity.cs

```diff
@@ -463,6 +470,26 @@ void Components_ComponentRemoved(Type t, Sandbox.Common.Components.MyComponentBa
             }
         }
 
+        public void AssignGamelogicFromHashSet(HashSet<Type> items)
+        {
+            bool first = true;
+            foreach (Type logicType in items)
+            {
+                if (first)
+                {
+                    GameLogic = (MyGameLogicComponent)Activator.CreateInstance(logicType);
```

Comment:
Hm, iteration order is unpredictable so it's unknown which component will end up in here.
Also, this single component can still be overwritten by setting GameLogic.